### ISSUE TITLE
QTabs ignores intended data-flow of v-model

### ISSUE THREAD

**Issue by renekaesler** [quasarframework/quasar#9063]
**Describe the bug**
When setting `:model-value` & `@update:model-value` explicitly, then model-value is updated, even if the updating method does not change the model.

This is a problem, when you want to interrupt tab switches based on a condition (e.g. an invalid form inside of the active tab panel)

**Codepen/jsFiddle/Codesandbox (required)**
https://codepen.io/pacman-rene-kaesler/pen/zYNXrpJ?editors=1010

**To Reproduce**
1. Click on an tab that is different  from mails

**Expected behavior**
Only if the event handler changes the model, the tab should switch.

**Platform (please complete the following information):**
Quasar Version:
@quasar/app Version:
Quasar mode:
  - [X] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

Tested on:
  - [X] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

OS: Ubuntu 20.04
Node: v14.15.0
NPM:6.14.8
Browsers: Google Chrome

**Comment by hawkeye64** [comment]
@renekaesler I think you are wanting to use the `keep-alive` property
https://next.quasar.dev/vue-components/tab-panels#usage

**Comment by metalsadman** [comment]
looks like a bug to me, model should not change, as `:model-value @update:model-value`, should behave like the old `:value @input`.

**Comment by renekaesler** [comment]
@hawkeye64 No, I is not about the `keep-alive` property. The codepen of @metalsadman is a good example for conditions in the update handler (thanks for the pen 👍🏼 ). 

Actually QTabs mirrors the `:model-value` in its own state ([`currentModel`](https://github.com/quasarframework/quasar/blob/c5a527b80de03fc40e03ab66f0a7afe3651a8fa7/ui/src/components/tabs/QTabs.js#L75)). I do not exactly know, why the model has been mirrored, but because of that you have to keep `:model-value` & `currentModel` in sync. You can find this logic inside the [`updateModel` method](https://github.com/quasarframework/quasar/blob/c5a527b80de03fc40e03ab66f0a7afe3651a8fa7/ui/src/components/tabs/QTabs.js#L152). But this logic is only executed, if the[ `onUpdate:modelValue`](https://github.com/quasarframework/quasar/blob/c5a527b80de03fc40e03ab66f0a7afe3651a8fa7/ui/src/components/tabs/QTabs.js#L157) has been set.

For me that does not seems to be right, because the logic assumes `v-model` has been set, when `model-value` & `update:model-value` is available in the props... This results in unusually behaviours:

- **`:model-value` has been set:** click on other tabs **does not** trigger a switch animation
- **`:model-value` & `update:model-value` (with empty handler) has been set:**  click on other tabs **does** trigger a switch animation

I think the problem lies in copying the `:model-value` into the components state. In most cases it is considered as an anti-pattern.

**Comment by jparish3** [comment]
what is the status of this open issue; seems fundamental to the component

**Comment by rstoenescu** [comment]
This was fixed in Quasar v2.9.0 (via d17978814): when an `@update:model-value` listener is present, clicking a tab now only emits the event and waits for your handler to write the value back, so rejecting the update (e.g. while a form is invalid) keeps the current tab selected. Verified against the latest release (v2.25.0): a handler that does not write back holds the tab, a conditional write-back accepts/rejects as expected, and with no listener QTabs self-manages (same convention as QDialog/QMenu/QDrawer). Please reopen if you can still reproduce it on the latest version.
